File: plugin/com.hello.uxml.tools.ui/src/com/hello/uxml/tools/framework/events/DataEvent.java
package com.hello.uxml.tools.framework.events;

/**
 * Defines data event arguments.
 * @author sanjayc@ (Sanjay Chouksey)
 */
public class DataEvent extends EventArgs {

  private Object data;

  /**
   * Constructor.
   */
  public DataEvent(Object source, Object data) {
    this.source = source;
    this.data = data;
  }

  /**
   * Gets or sets the data of the event.
   */
  public Object getData() {
    return data;
  }

  public void setData(Object data) {
    this.data = data;
  }
}

File: framework/lib/data_event.dart
part of uxml;

/**
 * Defines an event class that can include arbitrary data with the event.
 *
 * @author sanjayc@ (Sanjay Chouksey)
 */
class DataEvent extends EventArgs {

  /** The data associated with this event. */
  Object data;

  DataEvent(Object source, Object eventData) : super(source) {
    data = eventData;
  }
}
